.github/workflows/performance_tests.yml

```diff
@@ -0,0 +1,52 @@
+name: Performance Tests
+
+on:
+  pull_request:
+    branches:
+      - main
+  schedule:
+    - cron:  '0 9 * * 0'
+
+jobs:
+  check_branch_changes:
+    runs-on: ubuntu-22.04
+    outputs:
+      changed: ${{ steps.branch_change_output.outputs.has-new-commits }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: check branch changes
+        id: branch_change_output
+        uses: adriangl/check-new-commits-action@v1
+        with:
+          seconds: 604800
```

Comment:
Can you tell me a bit more about what this line is doing?

Reply:
We didn't want to run performance tests if there weren't any commits to dev. This line checks to see if there were commits to dev in the past week (or 604800 seconds) before continuing to run tests.